dhis-2/dhis-test-performance/src/test/org/hisp/dhis/test/tracker/TrackerTest.java

```diff
@@ -34,89 +34,325 @@
             .acceptHeader("application/json")
             .maxConnectionsPerHost(100)
-            .header("Content-Type", "application/json")
             .userAgentHeader("Gatling/Performance Test")
             .warmUp(
                 "http://localhost:8080/api/ping") // https://docs.gatling.io/reference/script/http/protocol/#warmup
-            .disableCaching(); // to repeat the same request without HTTP cache influence (304)
+            .disableCaching() // to repeat the same request without HTTP cache influence (304)
+            .check(status().is(200)); // global check for all requests
 
+    // only one user at a time
+    ScenarioWithRequests eventScenario = eventProgramScenario(repeat, eventProgram);
+    ScenarioWithRequests trackerScenario = trackerProgramScenario(repeat, trackerProgram);
+
+    List<Assertion> allAssertions = new java.util.ArrayList<>();
+    allAssertions.add(forAll().successfulRequests().percent().gte(100d));
+    allAssertions.addAll(eventScenario.requests().stream().map(Request::assertion).toList());
+    allAssertions.addAll(trackerScenario.requests().stream().map(Request::assertion).toList());
+
+    setUp(
+            eventScenario
+                .scenario()
+                .injectClosed(constantConcurrentUsers(1).during(1))
+                .andThen(
+                    trackerScenario.scenario().injectClosed(constantConcurrentUsers(1).during(1))))
+        .protocols(httpProtocolBuilder)
+        .assertions(allAssertions);
+  }
+
+  private ScenarioWithRequests eventProgramScenario(String repeat, String eventProgram) {
     String singleEventUrl = "/api/tracker/events/#{eventUid}";
     String relationshipUrl =
         "/api/tracker/relationships?event=#{eventUid}&fields=from,to,relationshipType,relationship,createdAt";
 
-    // get a 100 requests per run irrespective of the response times so comparisons are likely
-    // to be more accurate
     String getEventsUrl =
         "/api/tracker/events?program="
-            + program
-            + "&fields=dataValues,occurredAt,event,status,orgUnit,program,programType,updatedAt,createdAt,assignedUser,&orgUnit=DiszpKrYNg8&orgUnitMode=SELECTED&order=occurredAt:desc";
-
-    ScenarioBuilder scenario = scenario("Single Events");
-
-    scenario =
-        scenario
-            .exec(
-                http("Login")
-                    .post("/api/auth/login")
-                    .body(StringBody("{\"username\":\"admin\",\"password\":\"district\"}"))
-                    .check(status().is(200)))
+            + eventProgram
+            + "&fields=dataValues,occurredAt,event,status,orgUnit,program,programType,updatedAt,createdAt,assignedUser,"
+            + "&orgUnit=DiszpKrYNg8"
+            + "&orgUnitMode=SELECTED"
+            + "&order=occurredAt:desc";
+
+    Request goToFirstPage =
+        new Request(
+            getEventsUrl,
+            100,
+            "Go to first page of program " + eventProgram,
```

Comment:
Did you try only using the request name in the assertion? or is the fully qualified group + request name needed. I am asking because I think that's what we want and even though the knowledge of group + assertion is closer but its still not 100% tied together. We declare groups using the Gatling dsl and associate requests with it which we repeat here again.

Reply:
Unfortunately it doesn't work only with the request name.
I agree, the groups are now defined in one place and referenced somewhere else.
It can be improved but I didn't want to go too far at this point.